libvcx/src/aries/handlers/connection/connection.rs

```diff
@@ -350,13 +350,17 @@ Get messages received from connection counterparty.
     Sends authenticated message to connection counterparty
      */
     pub fn send_message(&self, message: &A2AMessage) -> VcxResult<()> {
-        trace!("Connection::send_message >>> message: {:?}", message);
+        let send_message = self.send_message_closure()?;
+        send_message(message)
+    }
```

Comment:
Does this function still need to be exposed?

Reply:
Yes
```
pub fn send_message(handle: u32, message: A2AMessage) -> VcxResult<()> {
    trace!("connection::send_message >>>");
    CONNECTION_MAP.get_mut(handle, |connection| {
        connection.send_message(&message)
    })
}
```
in `src/connection.rs`

Reply:
Can you not get the send message callback and call it instead?

Reply:
good point! Modified

Reply:
Is it possible to remove it now?